### DISCUSSION TITLE
feat: Support terraform `check` blocks

### DISCUSSION THREAD

**Discussion by nishigori** [aquasecurity/trivy#4671] [Ideas]
### Description

As of Terraform 1.5, support has been added for `check` blocks.

https://github.com/hashicorp/terraform/blob/v1.5/CHANGELOG.md?rgh-link-date=2023-06-16T21%3A56%3A02Z#150-june-12-2023
https://developer.hashicorp.com/terraform/language/checks

Today we do not support this, therefore scanning any file that has an `check` block will not be classified as a terraform config file.

```sh
$ trivy config .
2023-06-20T01:42:59.959+0900    INFO    Misconfiguration scanning is enabled
2023-06-20T01:43:00.285+0900    INFO    Detected config files: 0

$ cat main.tf
check "instance_running" {
  data "aws_instance" "example" {
    instance_id = "i-abcd1234"
  }

  assert {
    condition     = data.aws_instance.example.status == "running"
    error_message = "..."
  }
}
```

### Target

None

### Scanner

Misconfiguration

**Comment by nishigori** [comment]
Awaiting review for https://github.com/aquasecurity/defsec/pull/1354

**Reply by simar7** [reply]
Thank you. I will take a look at it.
